ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -597,8 +597,14 @@ + (void)updateViewController:(UIViewController *)vc
 #if !TARGET_OS_TV
           if (@available(iOS 11.0, *)) {
             RNSSearchBar *searchBar = subview.subviews[0];
-            navitem.searchController = searchBar.controller;
-            navitem.hidesSearchBarWhenScrolling = searchBar.hideWhenScrolling;
+            if(searchBar.removeTitle) {
+              [navitem.searchController.searchBar sizeToFit];
+              navitem.titleView = searchBar.controller.searchBar;
+              searchBar.controller.hidesNavigationBarDuringPresentation = false;
+            } else {
+              navitem.searchController = searchBar.controller;
+              navitem.hidesSearchBarWhenScrolling = searchBar.hideWhenScrolling;
```

Comment:
Shouldn't these options be set regardless of `searchBar.removeTitle` value? Even if the title is hidden, we want the search bar to be properly configured (fully functional).

Reply:
I've added hidWhenScrolling back , but searchController can't be assigned